espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/viewmatcher/WithXPath.java

```diff
@@ -1,23 +1,38 @@
 package io.appium.espressoserver.lib.viewmatcher;
 
+import android.support.annotation.Nullable;
 import android.view.View;
 
 import org.hamcrest.Description;
 import org.hamcrest.Matcher;
 import org.hamcrest.TypeSafeMatcher;
 
 
+import java.util.List;
+
 import io.appium.espressoserver.lib.handlers.exceptions.XPathLookupException;
 import io.appium.espressoserver.lib.model.SourceDocument;
 
 public class WithXPath {
-    public static Matcher<View> withXPath(final String xpath) throws XPathLookupException {
-        final View matchedXPathView = SourceDocument.findViewByXPath(xpath);
+    public static Matcher<View> withXPath(final String xpath, @Nullable final Integer index) throws XPathLookupException {
+
+        // Get a list of the Views that match the provided xpath
+        final List<View> matchedXPathViews = SourceDocument.findViewsByXPath(xpath);
 
         return new TypeSafeMatcher<View>() {
             @Override
             protected boolean matchesSafely(View item) {
-                return item.equals(matchedXPathView);
+                try {
+                    if (index != null) {
```

Comment:
nit: one says it's better to use positive comparisons rather than negative inside _if_ to improve readability (prefer `index == null` over `index != null`)